### PR TITLE
Task notes modals

### PR DESCRIPTION
**Pull request by TheHollidayInn**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes put_issue_url_here

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

This displays a model when the user clicks the notes icon on a task.

task-notes-popup

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

**Comment by khipkin** [comment]
@TheHollidayInn Can you please post some screenshots? 🙏

File: test/client-old/spec/directives/task.directive.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  function getCompiledElement(){
+    var element = angular.element('<task></task>');
+    var compiledElement = compile(element)(scope);
+    scope.$digest();
+    return compiledElement;
+  }
+
+  xit('opens task note modal', () => {
```
Comment (eastwood) [inline test/client-old/spec/directives/task.directive.js:30]: Does this need to be excluded?

Comment (TheHollidayInn) [reply test/client-old/spec/directives/task.directive.js:30]: Yes, we don't have a way to cache our jade templates to testing atm. And we have a new client coming up so this isn't super important.

**Comment by Alys** [comment]
- [ ] Emoji and markdown will need to be enabled in the modal before this goes live otherwise we'll get innumerable bug reports. :)

**Comment by TheHollidayInn** [comment]
Added markdown and emojis

**Review by paglias — APPROVED** [review]

**Pull request merged**